pkg/service/service.go

```diff
@@ -103,6 +103,25 @@ func (s *Service) Run() error {
 	}
 }
 
+func (s *Service) StartDebugHandler() {
+	if s.conf.DebugHandlerPort == 0 {
+		logger.Debugw("debug handler disabled")
+	}
+
+	h := &handlerProxyHandler{processManager: s.manager}
+
+	mux := http.NewServeMux()
+	mux.Handle(fmt.Sprintf("/%s/", gstPipelineDotFile), h)
+
+	go func() {
+		addr := fmt.Sprintf(":%d", s.conf.DebugHandlerPort)
+
+		logger.Debugw(fmt.Sprintf("starting debug handler on address %s", addr))
+		err := http.ListenAndServe(addr, mux)
+		logger.Infow("debug server failed", "error", err)
```

Comment:
ListenAndServe always returns a non-nil error, probably don't want to log it

Reply:
From my testing it doesn't return when the process/service gets killed, so we should not see the log message unless the server fails to start. I'll replace it with an Info message though